src/stream.iced

```diff
@@ -0,0 +1,56 @@
+stream = require('stream')
+enc = require('./encoding')
+
+desired_high_water_mark = 4096
+calculate_high_water_mark = (input_length) ->
+  input_length*Math.floor(desired_high_water_mark/input_length)
+
+exports.StreamEncoder = class StreamEncoder extends stream.Transform
+
+  constructor : (@encoder) ->
+    @extra = null
+    @block_size = @encoder.in_block_len
+    super({highWaterMark : calculate_high_water_mark(@block_size)})
```

Comment:
Do you even need to specify `highWaterMark` anymore?  It doesn't seem like it was buying us much since it wasn't doing what we thought it was doing anyways.